.github/workflows/test-and-deploy.yml

```diff
@@ -25,6 +25,10 @@ jobs:
           npm run test
           npm run ts-check
           python3 ./etc/scripts/util/propscheck.py
+          python3 ./etc/scripts/util/propschecktest.py
+          # TODO: use actions to set up python, use a separate virtual environment?
```

Comment:
To minimize this commit, I have decided to not use an [action](https://docs.github.com/en/actions/use-cases-and-examples/building-and-testing/building-and-testing-python#specifying-a-python-version) for python and not use a separate virtual environment.

Happy to change on request.